app/views/root/_gem_base.html.erb

```diff
@@ -32,6 +32,14 @@
   global_bar << global_banner if global_banner
 %>
 
+<% if ENV["GOOGLE_TAG_MANAGER_ID"] %>
+   <%= render "govuk_publishing_components/components/google_tag_manager_script", {
+     gtm_id: ENV["GOOGLE_TAG_MANAGER_ID"],
+     gtm_auth: ENV["GOOGLE_TAG_MANAGER_AUTH"],
+     gtm_preview: ENV["GOOGLE_TAG_MANAGER_PREVIEW"]
+   } %>
+ <% end %>
```

Comment:
Putting the component here means that it's placed before the opening `html` tag:
<img width="1166" alt="" src="https://user-images.githubusercontent.com/1732331/155114139-d54d3693-29ca-471c-b3e2-b539e7715c72.png">

For context, we're aiming to reduce our reliance on `static` for templates - the plan is for the templates to be used directly by a frontend application rather than from `static`. 

With this in mind, I'd suggest that this should be added as an option for the public layout component in the components gem, and either the environment variables used in the component if they can be access there or passed into the component parameters. 

Happy to talk this through if there are other pressures that I'm not aware of.

Reply:
Thanks @injms for your review. Can i just confirm that you mean [here](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_layout_for_public.html.erb)?

Reply:
```suggestion
<% content_for :head do %>
  <% if ENV["GOOGLE_TAG_MANAGER_ID"] %>
     <%= render "govuk_publishing_components/components/google_tag_manager_script", {
       gtm_id: ENV["GOOGLE_TAG_MANAGER_ID"],
       gtm_auth: ENV["GOOGLE_TAG_MANAGER_AUTH"],
       gtm_preview: ENV["GOOGLE_TAG_MANAGER_PREVIEW"]
     } %>
  <% end %>
<% end %>
```

I am suggesting we pass this script up to the layout component. I'd like to keep these details decoupled from the public layout and using env variables at the gem level doesn't sound like good practice. What do you think, @injms? Would you be ok with this approach?

Reply:
This way we can carry the GTM experiment with minimal impact on the existing component (as yield in `<head>` is already available).

Reply:
Passing it into the `:head` makes sense for this experiment - happy with that as an approach for this, but if it becomes more permanent I think it should be looked at again.

Reply:
OK great, thank you for @injms. I will dismiss your review where you requested the change, if you would mind approving please 😄